webui/react/src/shared/contexts/stores/UI.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { Dispatch, useContext, useReducer } from 'react';
+
+import rootLogger from 'shared/utils/Logger';
+import { checkDeepEquality } from 'shared/utils/store';
+
+const logger = rootLogger.extend('store');
+
+import { DarkLight, Mode, Theme } from '../../themes';
+
+interface StateUI {
+  chromeCollapsed: boolean;
+  darkLight: DarkLight;
+  isPageHidden: boolean;
+  mode: Mode;
+  showChrome: boolean;
+  showSpinner: boolean;
+  theme: Theme;
+}
+
+const initUI: StateUI = {
+  chromeCollapsed: false,
+  darkLight: DarkLight.Light,
+  isPageHidden: false,
+  mode: Mode.System,
+  showChrome: true,
+  showSpinner: false,
+  theme: {} as Theme,
+};
+
+enum StoreActionUI {
+  HideUIChrome = 'HideUIChrome',
+  HideUISpinner = 'HideUISpinner',
+  SetMode = 'SetMode',
+  SetPageVisibility = 'SetPageVisibility',
+  SetTheme = 'SetTheme',
+  ShowUIChrome = 'ShowUIChrome',
+  ShowUISpinner = 'ShowUISpinner',
+}
+
+type ActionUI =
+  | { type: StoreActionUI.HideUIChrome }
+  | { type: StoreActionUI.HideUISpinner }
+  | { type: StoreActionUI.SetMode; value: Mode }
+  | { type: StoreActionUI.SetPageVisibility; value: boolean }
+  | { type: StoreActionUI.SetTheme; value: { darkLight: DarkLight; theme: Theme } }
+  | { type: StoreActionUI.ShowUIChrome }
+  | { type: StoreActionUI.ShowUISpinner };
+
+class UIActions {
+  constructor(private dispatch: Dispatch<ActionUI>) {}
+
+  public hideChrome = (): void => {
+    this.dispatch({ type: StoreActionUI.HideUIChrome });
+  };
+
+  public hideSpinner = (): void => {
+    this.dispatch({ type: StoreActionUI.HideUISpinner });
+  };
+
+  public setMode = (mode: Mode): void => {
```

Comment:
these could use some jsdoc comments. I'll leave the motu for now. there is a rebase left to do